### PR TITLE
DDF-2071 Update CatalogFrameworkImpl so callers can supply a fully po…

### PR DESCRIPTION
**Pull request by glenhein**
#### What does this PR do?
Add the ability for the caller of CatalogFramework.create() to pass in a fully defined metacard instead of letting the framework create a metacard with the input transformers.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@stustison 
@jlcsmith 
@jrnorth 
@kcwire 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@jlcsmith
@kcwire
#### How should this be tested?
Run standard maven tests. There is no UI support for testing this change.

#### Any background context you want to provide?
This change is being made because the new UDP MPEG-TS Stream Monitor can  supply a fully detailed metacard more efficiently.

#### What are the relevant tickets?
DDF-2071

#### Checklist:
- [x] Documentation Updated (CatalogFramework javadoc)
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java [hunk 1]
```diff
@@ -867,13 +867,31 @@ private void generateMetacardAndContentItems(StorageRequest storageRequest,
...
+                Metacard metacard;
+
+                if (contentItem.getMetacard() != null) {
+                    metacard = contentItem.getMetacard();
+                    if (StringUtils.isBlank(metacard.getId()) ||
+                            StringUtils.isBlank(metacard.getTitle()) ||
+                            metacard.getAttribute(Metacard.POINT_OF_CONTACT) == null ||
+                            StringUtils.isBlank((String) metacard.getAttribute(Metacard.POINT_OF_CONTACT)
+                                    .getValue())) {
+                        throw new MetacardCreationException(
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java]: `String.format` or use a `StringBuilder` to minimize all this concatenation.

Comment (glenhein) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java]: changed to String.format

**Comment by coyotesqrl** [comment]
It appears there is no unit test to cover the new branch in `CatalogFrameworkImpl`.

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java [hunk 2]
```diff
@@ -867,13 +867,31 @@ private void generateMetacardAndContentItems(StorageRequest storageRequest,
...
-                        contentItem.getId(),
-                        fileName,
-                        size,
-                        (Subject) storageRequest.getProperties()
-                                .get(SecurityConstants.SECURITY_SUBJECT),
-                        tmpPath);
+
+                Metacard metacard;
+
+                if (contentItem.getMetacard() != null) {
```
Comment (stustison) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java]: This has me a little bit worried, the more I think about it. Currently, we use the generated metacard to determine whether or not the user and system has sufficient privileges to store that content item. With this change, I could now craft a metacard with any sort of security profile, and store any content item, regardless of content. 

We do allow you to craft any metacard and store any metacard in the catalog. I just feel like the difference with that, is you're storing the thing that we're checking. With this, you're storing a metacard along with a content item that could be entirely unrelated.

Comment (glenhein) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java]: How would we skip the input transformers?

**Comment by glenhein** [comment]
a added a unit test

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java (hunk 2 above)
Comment (stustison) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java]: That is sort of my point. We wouldn't be able to skip the input transformers.

**Comment by stustison** [comment]
In case it isn't clear from my previous comment and the discussions we've had: this is a 👎  until this is toggle-able in the framework and turned off by default. This description for the toggle should also have a nice warning about security.

**Pull request closed without merging**